### PR TITLE
Fixed lambda-uploader to work with Python 3, using tox for testing

### PR DESCRIPTION
**Pull request by jarosser06**
This change adds the ability to run lambda-uploader in a local Python 3
environment. The lambda-uploader will explicitly look for a python2
executable instead of just using python.

### CODE REVIEW

**Comment by jarosser06** [comment]
Resolves #43

File: circle.yml [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-
 machine:
   python:
-    version: 2.7
+    version: '2.7.10'
+
+dependencies:
+  override:
+    - pip install -U pip
+    - pip install -U tox tox-pyenv
```
Comment (martinb3) [inline circle.yml]: Should we also include tox/tox-pyenv in test-requirements.txt or setup.py?

File: circle.yml [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
+    - pyenv local 2.7.9 3.4.3 3.5.0
```
Comment (martinb3) [inline circle.yml]: Looks like there's a 3.5.1 out, as well as 2.7.11; should we use those as the latest? Or are these matching what CircleCI currently has?

File: tox.ini [hunk 3]
```diff
@@ -1,3 +1,22 @@
...
+commands=
+    python -c "import sys;print('\nPYTHON VERSION\n%s\n' % sys.version)"
+    py.test --verbose --color auto
+
+[testenv:style]
+deps =
+      .[style]
+basepython = python2.7
+commands =
+    flake8 lambda_uploader setup.py --statistics
```
Comment (martinb3) [inline tox.ini]: Did we want to go ahead and add pylint here?

File: circle.yml (hunk 2 above)
Comment (martinb3) [inline circle.yml]: @jarosser06 pointed me at https://circleci.com/docs/environment/#python, but it looks like we should bump to 2.7.10 at least.

Comment (theborch) [inline circle.yml]: @jarosser06 @martinb3 looks like if you use the trusty image you can have the latest: https://circleci.com/docs/trusty-image/

File: tox.ini (hunk 3 above)
Comment (jarosser06) [inline tox.ini]: Not in this run. There are a lot of changes, I'll make a separate PR for this to keep this one reasonable.

**Comment by martinb3** [comment]
After discussing with @jarosser06, since this PR and work was already done before we made our new procedures yesterday, I'm going to merge it and we'll get Pylint done ASAP.

**Pull request merged**